### PR TITLE
Replace free for all collation in `cumulus` runtimes

### LINKED ISSUE #103: Remove All "Free for All" Collator Selection
As part of block builder improvements, all parachains should have a configurable collator set (e.g. using Aura). Off the top of my head, this means changing Gluttons, but there may be some others (Penpal, Contracts, Template) that need changing too.

From @rphmeier :

> The "free for all" system does not work for building a backlog of blocks with more than a single collator.

### PR DESCRIPTION
**Pull request by georgepisaltu**
Partially fixes #103 

This PR removes instances of "free for all" collation in the `glutton`, `shell`, and `seedling` runtimes and replaces them with Aura instances. Aura is configured without a session manager, so the initial authority set cannot be changed later on.

### CODE REVIEW

**Comment by rphmeier** [comment]
This doesn't seem to achieve the goals of #103. 

1. Free-for-all collation is handled only on the client side and should not require anything in the runtime. It's not clear to me what the purpose of the new pallet is.
2. The glutton runtime is being updated to add both Aura and the new fixed pallet

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3548629

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3549585

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3549586

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3550250

**Comment by rphmeier** [comment]
> My understanding of https://github.com/paritytech/polkadot-sdk/issues/103 is that all runtimes running "free for all" (glutton, seedling and shell) should instead run with Aura.

Got it. We discussed two things in #103 and one of them was to build a new free-for-all implementation, so I thought this might've been about that.

I think the `fixed` pallet might be overkill, at least for glutton and tick - `pallet-aura` is already meant to be self-contained, at least for an unchanging authority set. The testing parachains definitely don't need that, but shell chains might really need authority set changes, will defer to @joepetrowski.

**Comment by BradleyOlson64** [comment]
This PR doesn't include changes I expected in
cumulus/polkadot-parachain/src/command.rs -> run().

Particularly, I think we'd want to change the glutton runtime case to `start_generic_aura_node` rather than `start_shell_node`

**Comment by georgepisaltu** [comment]
> I think the `fixed` pallet might be overkill, at least for glutton and tick - `pallet-aura` is already meant to be self-contained, at least for an unchanging authority set. The testing parachains definitely don't need that, but shell chains might really need authority set changes, will defer to @joepetrowski.

I spoke with @joepetrowski and it turns out we don't need to change the authority set in shell either, so I will drop the `fixed` pallet altogether and I'll move towards and impl where it's just Aura with no sessions.

**Comment by georgepisaltu** [comment]
> This PR doesn't include changes I expected in cumulus/polkadot-parachain/src/command.rs -> run().
> 
> Particularly, I think we'd want to change the glutton runtime case to `start_generic_aura_node` rather than `start_shell_node`

I don't understand exactly what you mean, is it related to the other thing in #103? For reference:
> Got it. We discussed two things in https://github.com/paritytech/polkadot-sdk/issues/103 and one of them was to build a new free-for-all implementation, so I thought this might've been about that.

I'm not familiar with the code you're pointing at, but judging by the name it seems a shell node would need to run Aura first in order to use `start_generic_aura_node`. To me it seems like the next step after the Aura switch.

**Comment by BradleyOlson64** [comment]
> I'm not familiar with the code you're pointing at, but judging by the name it seems a shell node would need to run Aura first in order to use `start_generic_aura_node`.

The impression I get is that by changing to `start_generic_aura_node` you are no longer running a shell node at all. It seems that since free for all collation isn't supported with Async Backing, we may want to retire shell node entirely. Though I don't know enough about shell vs generic aura to make that determination.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3607565

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3620186

**Review by BradleyOlson64 — APPROVED** [review]
Nice work

File: cumulus/parachains/runtimes/glutton/glutton-kusama/src/lib.rs
Comment (joepetrowski) [inline cumulus/parachains/runtimes/glutton/glutton-kusama/src/lib.rs:201]: I think this should be OK but maybe @Sophia-Gold can confirm. If OK then remove the TODO.

File: cumulus/parachains/runtimes/glutton/glutton-kusama/src/lib.rs
Comment (joepetrowski) [inline cumulus/parachains/runtimes/glutton/glutton-kusama/src/lib.rs:212]: For async backing, I think it needs to be this. However maybe this PR is just to add Aura and otherwise keep behavior the same.
```suggestion
	type SlotDuration = ConstU64<SLOT_DURATION>;
```

File: cumulus/parachains/runtimes/glutton/glutton-kusama/src/lib.rs
Comment (joepetrowski) [inline cumulus/parachains/runtimes/glutton/glutton-kusama/src/lib.rs:236]: Should be needed for Aura, yes
```suggestion
		Timestamp: pallet_timestamp::{Pallet, Call, Storage, Inherent} = 3,
```

File: cumulus/parachains/runtimes/glutton/glutton-kusama/src/lib.rs
Comment (joepetrowski) [inline cumulus/parachains/runtimes/glutton/glutton-kusama/src/lib.rs:246]: Changing indices is generally a no-no. Probably fine for Glutton since we don't have any in the wild. But what is the reason for this change?

**Review by joepetrowski — COMMENTED** [review]
Looks good, will approve once a few questions re async backing are answered. Comments on Glutton apply to all runtimes.

In `cumulus_pallet_parachain_system::Config`, I think you also need to change:

```rust
type CheckAssociatedRelayNumber = RelayNumberMonotonicallyIncreases;
type ConsensusHook = cumulus_pallet_aura_ext::FixedVelocityConsensusHook<
    Runtime,
    RELAY_CHAIN_SLOT_DURATION_MILLIS,
    BLOCK_PROCESSING_VELOCITY,
    UNINCLUDED_SEGMENT_CAPACITY,
>;
```

File: cumulus/parachains/runtimes/glutton/glutton-kusama/src/lib.rs
Comment (BradleyOlson64) [reply cumulus/parachains/runtimes/glutton/glutton-kusama/src/lib.rs:212]: The idea was to just add aura with this pr. I'll follow up with a pr containing async backing compatibility changes.

File: cumulus/parachains/runtimes/glutton/glutton-kusama/src/lib.rs
Comment (georgepisaltu) [reply cumulus/parachains/runtimes/glutton/glutton-kusama/src/lib.rs:236]: Yes I figured it out later but forgot to remove this comment.

File: cumulus/parachains/runtimes/glutton/glutton-kusama/src/lib.rs
Comment (georgepisaltu) [reply cumulus/parachains/runtimes/glutton/glutton-kusama/src/lib.rs:246]: I was shuffling them around while building the runtime and I didn't check the original indices when I posted the PR. Good catch, restored it now.

**Comment by georgepisaltu** [comment]
> Looks good, will approve once a few questions re async backing are answered. Comments on Glutton apply to all runtimes.
> 
> In `cumulus_pallet_parachain_system::Config`, I think you also need to change:
> 
> ```rust
> type CheckAssociatedRelayNumber = RelayNumberMonotonicallyIncreases;
> type ConsensusHook = cumulus_pallet_aura_ext::FixedVelocityConsensusHook<
>     Runtime,
>     RELAY_CHAIN_SLOT_DURATION_MILLIS,
>     BLOCK_PROCESSING_VELOCITY,
>     UNINCLUDED_SEGMENT_CAPACITY,
> >;
> ```

From the documentation of those items, I think you're right and it should be the way you describe above. I updated the config, PTAL.

File: cumulus/parachains/runtimes/glutton/glutton-kusama/src/lib.rs
Comment (Sophia-Gold) [reply cumulus/parachains/runtimes/glutton/glutton-kusama/src/lib.rs:211]: What's wrong with `SLOT_DURATION / 2`? I've used this in the upgraded tick and it works fine. Otoh, one would think 0 would be okay as well. I don't totally understand this, but anything less than `SLOT_DURATION` would seem to work.

Comment (georgepisaltu) [reply cumulus/parachains/runtimes/glutton/glutton-kusama/src/lib.rs:211]: There's nothing wrong with `SLOT_DURATION / 2`, let's leave it as is.

**Review by joepetrowski — APPROVED** [review]

**Pull request merged**
